CONTRIBUTING.md

```diff
@@ -153,13 +153,27 @@ This will run a local web server with "live reload" conveniently enabled. ( **NO
   git commit -s -m "<commit subject>"
   ```
 
-  **11.** Push the committed changes in your feature branch to your remote repo.
+  **11.** While you are working on your branch, other developers may update the `master` branch with their branch. This action means your branch is now out of date of the `master` branch and missing content. So in order to fetch the new changes, follow along:
+
+  ```
+  git checkout master
+  git fetch origin master
+  ```
+
+  Now you need to merge the `master` branch into your branch. This can be done in the following way:
+
+  ```
+  git checkout <your_branch_name>
+  git merge master
```

Comment:
I am not sure about the merge option, but rebase seems to be a better option ( widely used as well)

Reply:
Rebase is superb only if we don't open up a PR, if rebase is used in a PR then all files would appear (experienced this many times)

But for updating purposes, it is indeed a viable option

Reply:
Nope, that doesn't happen. I have been using rebase for almost 2.5yrs now and never experienced this issue if used correctly :)
If these commands work correctly, I don't have any issue in keeping them in the doc.

Reply:
@Nikhil-Ladha Yeah, They do work correctly, but what I have experienced is that when we do rebase, the commits which are accepted in the master get included in our PR. Maybe you could elaborate on what you said about "used correctly" or direct us to some resource so that when we use rebase. the commits do not get included in our PR.

Reply:
> @Nikhil-Ladha Yeah, They do work correctly, but what I have experienced is that when we do rebase, the commits which are accepted in the master get included in our PR. Maybe you could elaborate on what you said about "used correctly" or direct us to some resource so that when we use rebase. the commits do not get included in our PR.

Yup, cuz in my recent commit I just followed the existing commands which I included in the PR. I don't quite know the git best practices in depth so a resource will be great. Then I could change it :)

Reply:
+1 to @shubham-tyagi's Q

Reply:
Ok, so what I do is in case some commits got merged before my PR is merged, and I need those changes.
1. git checkout master
2. git fetch upstream
3. git merge upstream/master
4. git push
5. git checkout <development-branch>
6. git rebase master
7. Make my changes
8. git commit -s --amend or git commit -s -m "Commit msg"
9. git push --force origin <development branch>

These are the steps I follow and never face any issues.

Reply:
Thanks @Nikhil-Ladha !! I'll make the changes asap

Reply:
One question though, if we do ` git merge master` is it mandatory to add `--force` while doing the next push?

Reply:
Oh, then I would say let's keep the instructions you added here as it is.
Technically, from Git's POV, we shouldn't be using the `--force` argument while pushing but again its working compels us to do so. And, if there is a solution that avoids us from doing that let's use that solution.

Leaving it on you guys to decide, I have not used these commands so not sure about the exact steps, but will try it out soon.

Reply:
I guess `--force` is only mandatory when we have already pushed changes to our branch previously. like for example we did some work on our branch then pushed those commits to the branch. Then if in our machine we go like 2  commit backward by using `git reset` then we do some changes again in our branch then we would have to use `--force` because we have deleted the already pushed commit.

Reply:
But I don't think it is required to add `--force` in your case because `git checkout` is only used to eradicate the local changes. In that case, only `git push` should work.

Reply:
Not necessarily, if you do a `git commit --amend` and then do a push even then you would need to do `git push --force`.

Reply:
Yes, you are right @Nikhil-Ladha. I Forgot about amend 😬.

Reply:
so @Nikhil-Ladha @shubham-tyagi @alphaX86 shall I do the changes as Nikhil suggested here or keep it as it is?

> Ok, so what I do is in case some commits got merged before my PR is merged, and I need those changes.
> 
> 1. git checkout master
> 2. git fetch upstream
> 3. git merge upstream/master
> 4. git push
> 5. git checkout
> 6. git rebase master
> 7. Make my changes
> 8. git commit -s --amend or git commit -s -m "Commit msg"
> 9. git push --force origin
> 
> These are the steps I follow and never face any issues.

Reply:
I am fine with the `merge master` command as well.
It seems more convenient, else it causes unwanted trouble for new contributors.